src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -411,4 +405,67 @@ private void deleteFileAndInvokeCallback(
         mTotalSize = DiskBasedCacheUtility.removeEntry(key, mTotalSize, mEntries);
         callback.onGetComplete(null);
     }
+
+    /**
+     * Builder is used to build an instance of {@link DiskBasedAsyncCache} from values configured by
+     * the setters.
+     */
+    public static class Builder {
+        private FileSupplier rootDirectorySupplier;
```

Comment:
nit: Mark these as @Nullable

Reply:
I was getting an error when I did that, because it wanted the setter argument to be nullable as well. Should both be nullable, or is it better to leave the warning?